src/app/components/Tokens/TokenTransfers.tsx

```diff
@@ -0,0 +1,217 @@
+import { FC, useEffect, useState } from 'react'
+import { styled } from '@mui/material/styles'
+import { useTranslation } from 'react-i18next'
+import Box from '@mui/material/Box'
+import { Table, TableCellAlign, TableColProps } from '../Table'
+import { Layer, RuntimeEvent, useGetRuntimeEvmTokensAddress } from '../../../oasis-indexer/api'
+import { COLORS } from '../../../styles/theme/colors'
+import { TablePaginationProps } from '../Table/TablePagination'
+import { BlockLink } from '../Blocks/BlockLink'
+import { AccountLink } from '../Account/AccountLink'
+import { trimLongString } from '../../utils/trimLongString'
+import Typography from '@mui/material/Typography'
+import { TransactionLink } from '../Transactions/TransactionLink'
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward'
+import { TokenTransferIcon } from './TokenTransferIcon'
+import { RoundedBalance } from '../RoundedBalance'
+import { fromBaseUnits, getEthAccountAddressFromBase64, getEvmBech32Address } from '../../utils/helpers'
+import { AppErrors } from '../../../types/errors'
+
+const NULL_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+const iconSize = '28px'
+const StyledCircle = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  width: iconSize,
+  height: iconSize,
+  color: COLORS.eucalyptus,
+  backgroundColor: COLORS.lightGreen,
+  borderRadius: iconSize,
+  marginLeft: theme.spacing(3),
+  marginRight: `-${theme.spacing(4)}`,
+}))
+
+type TableRuntimeEvent = RuntimeEvent & {
+  markAsNew?: boolean
+}
+
+/**
+ * This is a wrapper around RoundedBalance which is able to find out the ticker by looking at an event
+ * @constructor
+ */
+const DelayedEventBalance: FC<
+  Omit<Exclude<Parameters<typeof RoundedBalance>[0], undefined>, 'ticker'> & {
+    event: RuntimeEvent
+  }
+> = props => {
+  const { t } = useTranslation()
+  const [oasisAddress, setOasisAddress] = useState<string | undefined>(undefined)
+  const { event, value, ...rest } = props
+
+  if (event.layer === Layer.consensus) {
+    throw AppErrors.UnsupportedLayer
+  }
+
+  const b64Address = event.body.address
+  const ethAddress = b64Address ? getEthAccountAddressFromBase64(b64Address) : undefined
+  useEffect(() => {
+    if (ethAddress) {
+      getEvmBech32Address(ethAddress).then(setOasisAddress)
+    }
+  }, [ethAddress])
+
+  const tokenQuery = useGetRuntimeEvmTokensAddress(event.network, event.layer, oasisAddress || '0x0')
```

Comment:
why `|| '0x0'`? that's making odd api request

Reply:
Leftover. I have meant to conditionally stop the request when we don't have data, but forgot to finish.

Reply:
Fixed. (We no longer call the API until we have the oasis address.)